rnaseq-cgl-pipeline/wrapper.py

```diff
@@ -1,102 +1,203 @@
-        if not all(x['Source'] == x['Destination'] for x in mounts):
-            raise IllegalArgumentException('Docker Src/Dst mount points, invoked with the -v argument,'
-                                           'must be the same if only using one mount point aside from the '
-                                           'docker socket.')
+        require(all(x['Source'] == x['Destination'] for x in mounts),
+                'Docker Src/Dst mount points, invoked with the -v argument, '
+                'must be the same if only using one mount point aside from the docker socket.')
         work_mount = [x['Source'] for x in mounts if 'docker.sock' not in x['Source']]
     else:
         # Ensure only one mirror mount exists aside from docker.sock
         mirror_mounts = [x['Source'] for x in mounts if x['Source'] == x['Destination']]
         work_mount = [x for x in mirror_mounts if 'docker.sock' not in x]
-        if len(work_mount) > 1:
-            raise IllegalArgumentException('Too many mirror mount points provided, see documentation.')
-        if len(work_mount) == 0:
-            raise IllegalArgumentException('No required mirror mount point provided, see documentation.')
-    # Enforce file input standards
+        require(len(work_mount) == 1, 'Wrong number of mirror mounts provided, see documentation.')
+    # Look for inputs in work directory
+    star = glob(os.path.join(work_mount[0], 'star*'))
+    rsem = glob(os.path.join(work_mount[0], 'rsem*'))
+    kallisto = glob(os.path.join(work_mount[0], 'kallisto*'))
+    if not args.star:
+        args.star = check_for_input(star, 'star')
+    if not args.rsem:
+        args.rsem = check_for_input(rsem, 'rsem')
+    if not args.kallisto:
+        args.kallisto = check_for_input(kallisto, 'kallisto')
+    # If sample is given as relative path, assume it's in the work directory
     if not all(x.startswith('/') for x in args.samples):
-        raise IllegalArgumentException("Sample inputs must point to a file's full path, e.g. "
-                                       "'/full/path/to/sample1.tar'. You provided {}.".format(args.samples))
-    if not all(x.startswith('/') for x in [args.star, args.kallisto, args.rsem]):
-        raise IllegalArgumentException("Sample inputs must point to a file's full path, e.g. "
-                                       "'/full/path/to/sample1.tar'. You  provided {}.".format(args.samples))
+        args.samples = [os.path.join(work_mount[0], x) for x in args.samples if not x.startswith('/')]
+        log.info('\nSample given as relative path, assuming sample is in work directory: {}'.format(work_mount[0]))
+    # Enforce file input standards
+    require(all(x.startswith('/') for x in args.samples),
+            "Sample inputs must point to a file's full path, "
+            "e.g. '/full/path/to/sample1.tar'. You provided {}.".format(args.samples))
+    require(all(x.startswith('/') for x in [args.star, args.kallisto, args.rsem]),
```

Comment:
No, since the pipeline will fail before this point if the one of the input args is missing.